generator/src/main/java/com/scottlogic/deg/generator/walker/ReductiveDecisionTreeWalker.java

```diff
@@ -64,14 +67,19 @@ public ReductiveDecisionTreeWalker(
         ReductiveState initialState = new ReductiveState(tree.fields);
         visualise(tree.getRootNode(), initialState);
         FixFieldStrategy fixFieldStrategy = fixFieldStrategyFactory.create(tree.getRootNode());
-        return fixNextField(tree.getRootNode(), initialState, fixFieldStrategy);
+        retryChecker.reset();
+        try {
+            return fixNextField(tree.getRootNode(), initialState, fixFieldStrategy);
+        } catch (RetryLimitReachedException e) {
+            return Stream.empty();
```

Comment:
It would be nice to have a message here to indicate to the user that no results have been found, and hence we are assuming no valid values exist.